src/catkin_pkg/condition.py

```diff
@@ -57,8 +57,13 @@ def _evaluate(parse_results, context):
         # return literal value
         return parse_results
 
+    assert len(parse_results) >= 3 and len(parse_results) % 2 == 1
     # recursion
-    assert len(parse_results) == 3
+
+    if len(parse_results) == 3:
+        parse_results = parse_results[:3]
+    else:
+        _evaluate(parse_results[2:], context)
```

Comment:
I don't think we need another case were `_evaluate()` is being called recursively. Instead the existing calls can be updated as described in https://github.com/ros-infrastructure/catkin_pkg/pull/279#issuecomment-596875191 to pass the right second argument.

Reply:
@dirk-thomas, I am unsure I got you correctly can you please specifically point which *existing calls* you mentioned to be updated? :smiley:

Reply:
https://github.com/ros-infrastructure/catkin_pkg/blob/dcd329d5291fc2bbe4b481f59f7eca7d9660cd75/src/catkin_pkg/condition.py#L64-L69